toolchain/sem_ir/typed_insts.h

```diff
@@ -795,6 +795,29 @@ struct FunctionTypeWithSelfType {
   InstId self_id;
 };
 
+// An overloaded C++ function declaration.
+struct OverloadedCppFunctionDecl {
+  static constexpr auto Kind =
+      InstKind::OverloadedCppFunctionDecl.Define<Parse::AnyFunctionDeclId>(
+          {.ir_name = "overloaded_cpp_fn_decl", .is_lowered = false});
+
+  TypeId type_id;
+  OverloadedCppFunctionId overloaded_function_id;
+};
+
+// The type of an overloaded C++ function.
+struct OverloadedCppFunctionType {
+  static constexpr auto Kind =
+      InstKind::OverloadedCppFunctionType.Define<Parse::AnyFunctionDeclId>(
```

Comment:
```suggestion
      InstKind::OverloadedCppFunctionType.Define<Parse::NodeId>(
```
This doesn't correspond to a Carbon `AnyFunctionDecl` parse node (or any Carbon parse node, really), so it doesn't make sense to pass a typed `NodeId` here.

Reply:
Done.